src/stats.rs

```diff
@@ -793,6 +1072,19 @@ impl StatValue {
     }
 }
 
+fn update_bucket_values(
+    bucket_values: &Vec<StatValue>,
+    buckets_to_update: &Vec<usize>,
+    change: &ChangeType,
+) {
+    for index in buckets_to_update.iter() {
```

Comment:
`.iter()` is normally unnecessary. Prefer `for x in &vec` - again Clippy should catch this?